### DISCUSSION TITLE
Allow setting the selected sub nav item in custom elements

### DISCUSSION THREAD

**Discussion by ryssbowh** [craftcms/cms#12680] [Ideas] [answered]
Hi,

It would be useful to be able to set which sub nav item should be selected when editing a custom element.
We have entry points to be able to change any other part of the edit page which is brilliant, that one seems to be missing ?

It's also achievable with something like this :
```
Event::on(Application::class, Application::EVENT_AFTER_REQUEST, function (Event $event) {
    $response = \Craft::$app->response;
    if ($response and $cpScreen = $response->getBehavior('cp-screen')) {
        if (StringHelper::startsWith($cpScreen->editUrl, 'my-custom/url')) {
            $cpScreen->selectedSubnavItem('my-custom-subnav');
        }
    }
});
```
But that feels way too much of a hack for something so simple.
It seems the only other alternative would be to rewrite the whole controller/template which is a shame since everything else is there already.

**Comment by brandonkelly** [comment, marked as answer]
You can do that via your element’s `prepareEditScreen()` method:

```php
use craft\web\CpScreenResponseBehavior;
use yii\web\Response;

// ...

public function prepareEditScreen(Response $response, string $containerId): void
{
    /** @var Response|CpScreenResponseBehavior $response */
    $response->selectedSubnavItem('my-custom-subnav');
}
```

**Comment by ryssbowh** [comment]
I missed that one. Thanks !
